client/src/components/SignupForm/SignupForm.js

```diff
@@ -1,49 +1,81 @@
 import React from 'react'
 import { connect } from 'react-redux'
-import { Field, reduxForm } from 'redux-form'
+import { Field, formValueSelector, reduxForm } from 'redux-form'
 import { push } from 'react-router-redux'
 import PropTypes from 'prop-types'
 import SignupValidator from './SignupValidator'
 
-import styles from '../LoginForm/LoginForm.scss'
+import styles from './SignupForm.scss'
 import { signup } from '../../actions'
 
-function EmailField ({ input, meta: { error } }) {
-  const emailClasses = `${styles.inputEmailContainer} ${error ? styles.error : styles.valid}`
+const EmailField = ({ input, meta: { dirty, error } }) => {
+  const emailClasses = `${styles.inputEmail} ${dirty ? error ? styles.error : styles.valid : ''}`
```

Comment:
Try to avoid using the ternary operator with nested conditionals especially when used within the same line as another computation due to it being a little complex to read. Try extracting out the second classname construction out of the template string and use a set of `if-else` instead.